rageval/metrics/_answer_f1.py

```diff
@@ -0,0 +1,158 @@
+import re
+import string
+from collections import Counter
+from dataclasses import dataclass
+from typing import List
+
+import datasets
+import numpy as np
+
+from rageval.metrics import Metric, add_attribute
+
+_DESCRIPTION = """\
+    F1 score combines precision and recall into a single score using their harmonic mean.
+"""
+
+_KWARGS_DESCRIPTION = """\
+Args:
+    name : str
+
+Optional Args:
+    None
+
+Functions:
+    _normalize_text: normalize the text by removing articles, white spaces, punctuations and lowercasing.
+    _validate_data: validate the dataset format.
+    _f1_score: compute the f1 score between `pred` string and `ref` string.
+    _compute_one: evaluate the f1 score of between `answer` and `gt_answers`, return the highest score in all pairs.
+
+Examples:
+    >>> from datasets import Dataset
+    >>> import rageval as rl
+    >>> sample = {
+    ...     "answers": [
+    ...         "Democrat rick kriseman won the 2016 mayoral election, while re- publican former mayor rick baker did so in the 2017 mayoral election."
+    ...     ],
+    ...     "gt_answers": [
+    ...         [
+    ...             "Kriseman",
+    ...             "Rick Kriseman"
+    ...         ]
+    ...     ]
+    ... }
+    >>> dataset = Dataset.from_dict(sample)
+    >>> metric = rl.metrics.AnswerF1Correctness()
+    >>> metric.mtype
+    'AnswerCorrectness'
+    >>> s, ds = metric.compute(dataset, batch_size=1)
+    >>> assert 0 <= s <= 1
+    >>> type(ds)
+    <class 'datasets.arrow_dataset.Dataset'>
+"""
+
+_CITATION = """\
+
+"""
+
+
+@dataclass
+@add_attribute('mtype', 'AnswerCorrectness')
+@datasets.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
+class AnswerF1Correctness(Metric):
+    """Estimates the F1 between answers and ground truth answers."""
+
+    name = "answer_f1"
+
+    ALIAS = ['answer_f1']
+
+    def __init__(self):
+        """
+        Explicitly initialize AnswerF1Correctness.
+
+        Ensure all parent classes are initialized.
+        """
+        super().__init__()
+        self._required_columns = ['answers', 'gt_answers']
+
+    def __repr__(self) -> str:
+        """:return: Formatted string representation of the metric."""
+        return f"{self.ALIAS[0]}"
+
+    def _info(self):
```

Comment:
`reference_urls` and `codebase_urls` is missing

Reply:
done